src/server/api/routers/feedback.ts

```diff
@@ -0,0 +1,64 @@
+import { z } from 'zod'
+import * as schema from '~/server/db/migration/schema'
+
+import { createTRPCRouter, integrationProcedure } from '~/server/api/trpc'
+import { db } from '~/server/db'
+import { FeedbackTypeEnum } from '~/model/feedback'
+import { uuidv7 } from 'uuidv7'
+import { asc, eq, sql } from 'drizzle-orm'
+
+export const createFeedback = createTRPCRouter({
+  createRating: integrationProcedure
+    .input(
+      z.object({
+        threadId: z.string(),
+        chatId: z.string(),
+        feedbackType: z.nativeEnum(FeedbackTypeEnum),
+        notes: z.string().optional(),
+      }),
+    )
+    .mutation(async ({ input, ctx }) => {
+      const feedbackId = uuidv7()
+      await db
+        .insert(schema.chat_feedback)
+        .values({
+          id: feedbackId,
+          threadId: input.threadId,
+          chatId: input.chatId,
+          typeId: input.feedbackType,
+          notes: input.notes,
+          createdAt: new Date(),
+        })
+        .returning()
+    }),
+
+  getList: integrationProcedure
+    .input(
+      z.object({
+        threadId: z.string(),
+        limit: z.number().max(100).default(10),
+        offset: z.number().default(0),
+      }),
+    )
+    .query(async ({ input }) => {
+      const countRow = await db
+        .select({
+          count: sql<number>`COUNT(*)`,
+        })
+        .from(schema.chat_feedback)
+        .where(eq(schema.chat_feedback.threadId, input.threadId))
+      const count = Number(countRow[0]?.count) ?? 0
+
+      const chat_feedback = await db.query.chat_feedback.findMany({
+        where: eq(schema.chat_feedback.threadId, input.threadId),
+        orderBy: [asc(schema.chat_feedback.id)],
```

Comment:
This is getting the oldest feedbacks first. 
Could consider to get latest first or not ?
Or the FE need sort options, we could consider support sending order from FE as well.